datasets/cms_medicare/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,174 @@
+    download_file(source_url, source_file)
+
+    logging.info(f"Opening file {source_file}")
+
+    if (
+        pipeline_name == "inpatient_charges_2011"
+        or pipeline_name == "inpatient_charges_2012"
+        or pipeline_name == "inpatient_charges_2013"
+        or pipeline_name == "inpatient_charges_2014"
+        or pipeline_name == "inpatient_charges_2015"
+        or pipeline_name == "outpatient_charges_2012"
+        or pipeline_name == "outpatient_charges_2013"
+        or pipeline_name == "outpatient_charges_2014"
+        or pipeline_name == "outpatient_charges_2011"
+    ):
+
+        with ZipFile(source_file) as zipped_files:
+
+            file_list = zipped_files.namelist()
+            csv_file = fnmatch.filter(file_list, "*.csv")
+            data = zipped_files.open(*csv_file)
+            df = pd.read_csv(data)
+
+    else:
+
+        df = pd.read_csv(str(source_file))
+
+    logging.info(f"Transformation Process Starting.. {source_file}")
+
+    logging.info(f"Transform: Renaming Headers.. {source_file}")
+    rename_headers(df, rename_mappings)
+
+    logging.info(f"Transform: Filtering null identity records.. {source_file}")
+    filter_null_rows(df, pipeline_name)
+
+    logging.info("Transform: Reordering headers..")
+    df = df[headers]
+
+    logging.info(f"Transformation Process complete .. {source_file}")
+
+    logging.info(f"Saving to output file.. {target_file}")
+
+    try:
+        save_to_new_file(df, file_path=str(target_file))
+    except Exception as e:
+        logging.error(f"Error saving output file: {e}.")
+
+    logging.info(
+        f"Uploading output file to.. gs://{target_gcs_bucket}/{target_gcs_path}"
+    )
+    upload_file_to_gcs(target_file, target_gcs_bucket, target_gcs_path)
+
+    logging.info(
+        "CMS Medicare process completed at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+
+def convert_dt_format(date_str: str, time_str: str) -> None:
+    return str(datetime.datetime.strptime(date_str, "%m/%d/%Y").date()) + " " + time_str
+
+
+def rename_headers(df: pd.DataFrame, rename_mappings: dict) -> None:
+    df.rename(columns=rename_mappings, inplace=True)
+
+
+def filter_null_rows(df: pd.DataFrame, pipeline_name: str):
+    if (
+        pipeline_name == "inpatient_charges_2011"
+        or pipeline_name == "inpatient_charges_2012"
+        or pipeline_name == "inpatient_charges_2013"
+        or pipeline_name == "inpatient_charges_2014"
+        or pipeline_name == "inpatient_charges_2015"
+    ):
+
+        return df.dropna(subset=["drg_definition", "provider_id"], inplace=True)
+
+    elif (
+        pipeline_name == "outpatient_charges_2012"
+        or pipeline_name == "outpatient_charges_2013"
+        or pipeline_name == "outpatient_charges_2014"
+        or pipeline_name == "outpatient_charges_2011"
+    ):
+
+        return df.dropna(subset=["apc", "provider_id"], inplace=True)
+
+    else:
+
+        df = df
```

Comment:
This seems redundant. Do you mean `return df`?

Reply:
Yes, I have added return df.